linera-client/src/client_context.rs

```diff
@@ -583,135 +574,73 @@ where
+            .make_benchmark_chains(num_chains, tokens_per_chain)
+            .await?;
+        info!(
+            "Got {} chains in {} ms",
+            key_pairs.len(),
+            start.elapsed().as_millis()
+        );
 
-            // Replay the certificate locally.
-            // No required certificates from other chains: This is only used with benchmark.
-            chain_client
-                .process_certificate(certificate.clone())
+        if let Some(id) = fungible_application_id {
+            let start = Instant::now();
+            self.supply_fungible_tokens(&key_pairs, id, &chain_clients)
                 .await?;
-
-            num_sent_proposals += 1;
-            if let Some(bps) = bps {
-                if num_sent_proposals == bps {
-                    let elapsed = start.elapsed();
-                    if elapsed > Duration::from_secs(1) {
-                        warn!(
-                            "Failed to achieve {} BPS/{} TPS, took {} ms",
-                            bps,
-                            bps * transactions_per_block,
-                            elapsed.as_millis()
-                        );
-                    } else {
-                        tokio::time::sleep(Duration::from_secs(1) - elapsed).await;
-                        info!(
-                            "Achieved {} BPS/{} TPS in {} ms",
-                            bps,
-                            bps * transactions_per_block,
-                            elapsed.as_millis()
-                        );
-                    }
-                    start = Instant::now();
-                    num_sent_proposals = 0;
-                }
-            }
-        }
-
-        if bps.is_none() {
-            let elapsed = start.elapsed();
-            let bps = num_sent_proposals as f64 / elapsed.as_secs_f64();
             info!(
-                "Achieved {} BPS/{} TPS",
-                bps,
-                bps * transactions_per_block as f64
+                "Supplied fungible tokens in {} ms",
+                start.elapsed().as_millis()
             );
         }
 
-        self.close_benchmark_chains(chain_clients).await?;
-        Ok(())
+        let default_chain_id = self
+            .wallet
+            .default_chain()
+            .expect("should have default chain");
+        let default_chain_client = self.make_chain_client(default_chain_id)?;
+        let (epoch, committees) = default_chain_client
+            .epoch_and_committees(default_chain_id)
+            .await?;
+        let epoch = epoch.expect("default chain should have an epoch");
+        let committee = committees
+            .get(&epoch)
+            .expect("current epoch should have a committee");
+        let blocks_infos = self.make_benchmark_block_info(
+            key_pairs,
+            transactions_per_block,
+            fungible_application_id,
+        );
+
+        Ok((chain_clients, epoch, blocks_infos, committee.clone()))
```

Comment:
This method is doing at least four things:
- clearing out inboxes, creating blocks
- creating new chains for benchmark
- supplying a test application with some tokens
- calling `make_benchmark_block_info` on a default chain
-